### PR TITLE
ROX-15101 nack to compliance if central connection is not ready

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

The idea is to have a mechanism to notify `SensorComponents` when the connection with central is established.

I'm thinking of making the `NotifyReady` function more generic and have something like `Notify(msg)` so we can reuse this for other types of notifications and not just notifying when the connection with central is ready.

Note to reviewers: Consider reviewing the PR by commit. The second and forth are the most interesting 🙂 

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

* Tested sending node inventories on sensor's startup with a broken connection.

### CODE REVIEW

File: sensor/common/component.go [hunk 1]
```diff
@@ -6,11 +6,20 @@ import (
 	"google.golang.org/grpc"
 )
 
+// SensorComponentEvent represents events about which sensor components can be notified
+type SensorComponentEvent string
+
+const (
+	// SensorComponentEventCentralReachable denotes that Sensor-Central connection is up
+	SensorComponentEventCentralReachable SensorComponentEvent = "central-reachable"
```
Comment (Maddosaurus) [inline sensor/common/component.go:14]: Optional nit: Though this is a very expressive var name, it is very long.
I struggle to come up with some shorter ones as well, but how about something like `CentralReached` or `CentralContacted`?

**Review by Maddosaurus — APPROVED** [review]
I see you've gone for a more generic Notify function, very cool.
The nit is optional, just a thought that ocured to me.

**Review by vikin91 — COMMENTED** [review]
Ups, apparently I forgot to submit my review comments. Here they are (all of them are addressed currently).

**Review by jschnath — APPROVED** [review]
LGTM

**Pull request merged**